### ISSUE TITLE
How do I #each loop through all top-level resource routes?

### ISSUE THREAD

**Issue by DanielJRaine** [ga-wdi-boston/capstone-project#290]
I am trying to figure out how to populate a nav-bar component template with all available resource routes without having to hard code each one (I have many nested nav-bars, and I want to be able to do this for each nested nav-bar so I do not have to keep hard-coding `#link-to` statements every time I add a route or nested route.)

Is there a way to get access to all available routes?  I'm also not sure how I would make this information available to my #each helper method.

**Comment by jrhorn424** [comment]
I completely understand your desire to link your UI to your route structure. This is another example of what I'd call "incidental duplication", or where our clean code instincts can start to betray us.

Consider:
-   You may not want every route in the navbar at some point in the future.
-   You may not want your link text to follow the name of the route as defined in the router.
-   You may not want to link your navbar to a private API in Ember that may change in the future.

If you want to try it anyway, the proper and safe way to proceed would be to only use methods publicly available on Ember Router (http://emberjs.com/api/classes/Ember.Router.html). I didn't see any methods, public or private, that would help, but that doesn't mean they're not there.

This is ultimately your call, but I'm not entirely on board FWIW.

**Comment by DanielJRaine** [comment]
Well, I don't feel quite as bad about hand-writing all of my #link-to's, but I would still like to know how to get a an array or object with all available routes if it is possible.

**Comment by jrhorn424** [comment]
>  but I would still like to know how to get a an array or object with all available routes if it is possible.

I'd be interested in knowing whether that's possible as well. The API docs don't seem to indicate a way, but I didn't deep dive any source code.

What have you found in your research so far?

**Comment by DanielJRaine** [comment]
A lot that has my head-spinning, and I have not found anyone trying to do the same thing. I'm stepping away from the problem until I make progress elsewhere, but once I find something that actually gets me anywhere, I'll post it here.

Pending thoughts:
If I want my model, route, and displayed link names to all be the same thing (I do for this project, for the time being), then I could maybe use some sort of find method in the store to return all models, and use the model names to build up URLs?  (this feels a bit hack-and-slashy)

**Comment by DanielJRaine** [comment]
Solved.  In this case, I have nested routes that follow the form /<gallery-name>/<exhibit-name>
I haven't done this for the top level resource routes yet, but for the 'Visual' gallery (first child route) I have used the model passed to it ('exhibit') in an #each loop using the concat helper function to build a my #link-to string.

This creates a list of all of the routes corresponding to each 'exhibit' model.  I should be able to do this for top-level routes too, which is what I'll try next.  

```
<h2>This is the Visual Gallery</h2>
<ul>
    {{#each model as |exhibit|}}
      <li>
        {{#link-to (concat "visual." exhibit.name)}}
          {{exhibit.name}}
        {{/link-to}}
      </li>
    {{/each}}
</ul>
```
